crates/types/src/config/cli_option_overrides.rs

```diff
@@ -82,6 +82,13 @@ pub struct CommonOptionCliOverride {
     #[clap(long, global = true)]
     pub bootstrap_num_partitions: Option<NonZeroU64>,
 
+    /// # Automatically provision number of configured partitions
+    ///
+    /// If this option is set to `false`, then one needs to manually write a partition table to
+    /// the metadata store. Without a partition table, the cluster will not start.
+    #[clap(long, global = true)]
+    pub auto_provision_partitions: Option<bool>,
```

Comment:
I think we will need to review the ergonomics of this process but I wouldn't block merging this.

Reply:
Sounds good :-)